src/main/java/org/openrewrite/java/spring/boot3/RemoveSolrAutoConfigurationExclude.java

```diff
@@ -0,0 +1,95 @@
+    @Override
+    public String getDescription() {
+        return "`SolrAutoConfiguration` was removed in Spring Boot 3; remove references to it from exclusions on annotations.";
+    }
+
+    private static final String SPRING_BOOT_APPLICATION = "org.springframework.boot.autoconfigure.SpringBootApplication";
+    private static final String ENABLE_AUTO_CONFIGURATION = "org.springframework.boot.autoconfigure.EnableAutoConfiguration";
+    private static final AnnotationMatcher SBA_MATCHER = new AnnotationMatcher(SPRING_BOOT_APPLICATION);
+    private static final AnnotationMatcher EAC_MATCHER = new AnnotationMatcher(ENABLE_AUTO_CONFIGURATION);
+
+    private static final String SOLR_AUTO_CONFIGURATION = "SolrAutoConfiguration";
+    private static final String SOLR_AUTOCONFIGURATION_FQN = "org.springframework.boot.autoconfigure.solr." + SOLR_AUTO_CONFIGURATION;
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.or(
+                        new UsesType<>(SPRING_BOOT_APPLICATION, true),
+                        new UsesType<>(ENABLE_AUTO_CONFIGURATION, true)),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+                        J.Annotation a = super.visitAnnotation(annotation, ctx);
+                        if (!SBA_MATCHER.matches(a) && !EAC_MATCHER.matches(a)) {
+                            return a;
+                        }
+                        maybeRemoveImport(SOLR_AUTOCONFIGURATION_FQN);
+
+                        return a.withArguments(ListUtils.map(a.getArguments(), it -> {
+                            if (it instanceof J.Assignment) {
+                                J.Assignment as = (J.Assignment) it;
+                                if (as.getAssignment() == null || !"exclude".equals(((J.Identifier) as.getVariable()).getSimpleName())) {
+                                    return it;
+                                }
+                                if (isSolrAutoConfigurationClassReference(as.getAssignment())) {
+                                    return null;
+                                } else if (as.getAssignment() instanceof J.NewArray) {
+                                    J.NewArray array = (J.NewArray) as.getAssignment();
+                                    List<Expression> newInitializer = ListUtils.map(array.getInitializer(),
+                                            expr -> isSolrAutoConfigurationClassReference(expr) ? null : expr);
+                                    //noinspection DataFlowIssue
+                                    if (newInitializer.isEmpty()) {
+                                        return null;
+                                    }
+                                    return maybeAutoFormat(it, as.withAssignment(array.withInitializer(newInitializer)), ctx);
```

Comment:
@Laurens-W This quick method allows you to apply the detected style to a code element, such that you need not worry about carrying over a specific space here or there.